app/services/user_subscriptions/subscription_cache_checker.rb

```diff
@@ -0,0 +1,28 @@
+module UserSubscriptions
+  class SubscriptionCacheChecker
+    attr_accessor :user, :source_type, :source_id
+
+    def self.call(*args)
+      new(*args).call
+    end
+
+    def initialize(user, source_type, source_id)
+      @user = user
+      @source_type = source_type
+      @source_id = source_id
+    end
+
+    def call
+      return false unless user
```

Comment:
I wonder if `return`ing `false` here is the right thing to do; if there is no `user` passed, then this would return `false`, and then `is_subscribed` would [be `false`](https://github.com/thepracticaldev/dev.to/pull/8779/files#diff-2ee6f8c98b7e8296071384bd2ee06ac9R11). But in reality, it's not that the user is not subscribed, it's that we didn't pass the `SubscriptionCacheChecker` a `user`... 🤔

Should we be handling the case where no `user` is passed to the `SubscriptionCacheChecker` class differently? Raise an error and then notify that we tried to call this service without a user?

Reply:
I'm torn on this. This case shouldn't be hit since we authenticate first. So this would happen if we authenticated and `current_user` was `nil`. By returning `false` it's just informing the client to treat the UX as if the user is not already subscribed. The client will already have the logic of its own to detect if a user is signed in or not as well which would supersede anything related to subscribed or not 🤔 

If I returned an error, instead, I'm not sure what that would look like on the client. That would be a "you're signed in, but not really" scenario 😅 . Maybe what I'd have to do is another confirmation that `current_user` exists before calling this and remove that guard? Is that overkill? What do you think?

Reply:
> Maybe what I'd have to do is another confirmation that current_user exists before calling this and remove that guard? Is that overkill? What do you think?

This feels right to me; seems like the controller can just handle this and the cache checker class shouldn't need to every worry about it at all. And actually, if `UserSubscriptionsController` inherits from `ApplicationController`, then you should be able to use `authenticate_user!` to check that the `current_user` exists and then remove this guard completely!

Reply:
Great point - I'll just drop the guard 🎉